File: lib/environment_config.dart
import 'dart:io';

enum Environment {
  dev,
  test,
  prod,
}

class EnvironmentConfig {
  static Environment environment = Environment.dev;  // Default environment

  static String get baseUrl {
    if (Platform.isAndroid) {
      switch (environment) {
        case Environment.dev:
          return 'http://10.0.2.2:3000';
        case Environment.test:
          return 'https://test.spacemate.io';
        case Environment.prod:
          return 'https://spacemate.io';
      }
    } else if (Platform.isIOS) {
      switch (environment) {
        case Environment.dev:
          return 'http://localhost:3000';
        case Environment.test:
          return 'https://test.spacemate.io';
        case Environment.prod:
          return 'https://spacemate.io';
      }
    }
    return 'http://127.0.0.1:3000';
  }
}

File: native/android/app/src/main/java/com/spacemate/app/config/EnvironmentConfig.kt
package com.spacemate.app.config

enum class Environment {
    DEV, TEST, PROD
}

object EnvironmentConfig {
    var environment: Environment = Environment.PROD

    val baseUrl: String
        get() = when (environment) {
            Environment.DEV -> "http://10.0.2.2:3000" // Android emulator localhost
            Environment.TEST -> "https://test.spacemate.io"
            Environment.PROD -> "https://spacemate.io"
        }
} 